src/OrbitGl/ThreadTrack.h

```diff
@@ -57,6 +58,8 @@ class ThreadTrack final : public TimerTrack {
 
   [[nodiscard]] std::vector<CaptureViewElement*> GetVisibleChildren() override;
 
+  void OnCaptureComplete();
```

Comment:
I would receive the flag scope_tree_update_type_  here instead of receiving it in the constructor.

Reply:
The flag is needed earlier so that we can avoid constructing the ScopeTree as timers arrive in the case of capture loading.

Reply:
See ThreadTrack::OnTimer

Reply:
Hmm, I might have missed a file when splitting up my PR... :-) Pushing now...

Reply:
It's in now, PTAL, that was an important piece...

Reply:
ahhh, nice to see. BTW, I'm still not sure. I will re-read again on Monday if you don't merge it today.